### PR TITLE
plugins(NormalizeResourceLinks): create plugin

### PR DESCRIPTION
**Pull request by bb010g**
Finally, I can stop manually removing the `canary.` from my copied links

This patches the copy link button action (which directly reads ambient `location.host`), because any deeper patches would risk breaking functionality.

### CODE REVIEW

**Comment by TheKodeToad** [comment]
Can't you use TextReplace?

**Comment by TheKodeToad** [comment]
Never mind. I'm stupid.

File: src/plugins/normalizeResourceLinks/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export default definePlugin({
+    name: "NormalizeResourceLinks",
+    description: "Normalize resource links to match stable Discord",
+    authors: [Devs.bb010g],
+    patches: [
+        {
+            find: ".Messages.COPY_MESSAGE_LINK,",
+            replacement: {
+                match: /(""\.concat\(location\.protocol,"\/\/"\)\.concat\()(location\.host)(\)\.concat\()/,
```
Comment (Vendicated) [inline src/plugins/normalizeResourceLinks/index.ts:18]: this matcher seems overly complex. You just need to replace location.host, right? Something like this should work fine

```js
match: /\.concat\(location\.host\)/,
replace: ".concat(location.host.replace(/^(canary|ptb)\\./, ''))"
```

Comment (bb010g) [reply src/plugins/normalizeResourceLinks/index.ts:18]: Is this safe to do? I don't want to mess up other code that depends on the actual `location.host` value.

Comment (Vendicated) [reply src/plugins/normalizeResourceLinks/index.ts:18]: the find limits it to only the module you want to match, which in this case is literally just the element and nothing else
image

File: src/plugins/normalizeResourceLinks/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    name: "NormalizeResourceLinks",
+    description: "Normalize resource links to match stable Discord",
+    authors: [Devs.bb010g],
+    patches: [
+        {
+            find: ".Messages.COPY_MESSAGE_LINK,",
+            replacement: {
+                match: /(\.concat\()(location\.host)(\))/,
+                replace:
+                    "$1Vencord.Plugins.plugins.NormalizeResourceLinks.normalizeResourceHost($2)$3",
```
Comment (Vendicated) [inline src/plugins/normalizeResourceLinks/index.ts:20]:
```suggestion
                    "$1$self.normalizeResourceHost($2)$3",
```

**Comment by Vendicated** [comment]
try to avoid redundant groups, it makes patches way harder to read. just repeat short constant strings or use lookarounds

**Comment by Vendicated** [comment]
thank you

**Pull request merged**